integration/spark/src/main/scala/org/apache/carbondata/spark/rdd/Compactor.scala

```diff
@@ -69,7 +69,9 @@ object Compactor {
       schemaName,
       factTableName,
       validSegments,
-      carbonTable.getAbsoluteTableIdentifier.getCarbonTableIdentifier.getTableId
+      carbonTable.getAbsoluteTableIdentifier.getCarbonTableIdentifier.getTableId,
+      colCardinality = Array[Int](0),
```

Comment:
Instead of creating an empty list and reassigning it you can create a reference of Array[Int] type like
var colCardinality: Array[Int] = null